content/protocols/sse.textile

```diff
@@ -0,0 +1,228 @@
+
+Read more in the "delta section":/channels/options/delta.
+
+h3(#delta-example). Delta example with SSE
+
+You can subscribe to messages in delta mode, using the SSE transport, as follows.
+
+```[javascript](code-editor:realtime/channel-deltas-sse-enveloped)
+  /* Make sure to include <script src="https://cdn.ably.com/lib/delta-codec.min-1.js"></script> in your head */
+  var key = '{{API_KEY}}';
+  var channel = '{{RANDOM_CHANNEL_NAME}}';
+  var baseUrl = 'https://realtime.ably.io/event-stream';
+  var urlParams = `?channels=${channel}&v=1.1&key=${key}&delta=vcdiff`;
+  var url = baseUrl + urlParams;
+  var eventSource = new EventSource(url);
+  var channelDecoder = new DeltaCodec.CheckedVcdiffDecoder();
+
+  eventSource.onmessage = function(event) {
+    /* event.data is JSON-encoded Ably Message
+       (see https://ably.com/docs/realtime/types#message) */
+    var message = JSON.parse(event.data);
+    var { id, extras } = message;
+    var { data } = message;
+
+    try {
+      if (extras && extras.delta) {
+        data = channelDecoder.applyBase64Delta(data, id, extras.delta.from).asUtf8String();
+      } else {
+        channelDecoder.setBase(data, id);
+      }
+    } catch(e) {
+      /* Delta decoder error */
+      console.log(e);
+    }
+
+    /* Process decoded data */
+    console.log(data);
+  };
+```
+
+h3(#delta-example-unenv). Delta example with unenveloped SSE
+
+For more information on enveloped and unenveloped SSE, please see the "SSE API":/api/sse#sse
+
+```[javascript](code-editor:realtime/channel-deltas-sse)
+  /* Make sure to include <script src="https://cdn.ably.com/lib/delta-codec.min-1.js"></script> in your head */
+  var DeltaCodec = require('@ably/delta-codec');
+
+  var key = '{{API_KEY}}';
+  var channel = 'sample-app-sse';
+  var baseUrl = 'https://realtime.ably.io/event-stream';
+  var urlParams = `?channels=${channel}&v=1.1&key=${key}&delta=vcdiff&enveloped=false`;
+  var url = baseUrl + urlParams;
+  var eventSource = new EventSource(url);
+  var channelDecoder = new DeltaCodec.VcdiffDecoder();
+
+  eventSource.onmessage = function(event) {
+      var data = event.data;
+
+      try {
+          if (DeltaCodec.VcdiffDecoder.isBase64Delta(data)) {
+              data = channelDecoder.applyBase64Delta(data).asUtf8String();
+          } else {
+              channelDecoder.setBase(data);
+          }
+      } catch(e) {
+          /* Delta decoder error */
+          console.log(e);
+      }
+
+      /* Process decoded data */
+      console.log(data);
+  };
+```
+
+h3(#rewind-sse). Rewind with SSE
+
+To use rewind with SSE you specify the @rewind@ channel option with a numerical value equal to the position you want to attach to, for example @"1"@ to rewind 1 message previous.This will apply to all channels specified in the query string.
```

Comment:
```suggestion
The @rewind@ channel option enables a client to specify where to start an attachment from. This can be a point in time in the past, or a given number of messages.
```